src/modals/ForceUndelegateModal.tsx

```diff
@@ -0,0 +1,263 @@
+    onResolve?: (sponsorshipId: string) => void
+    onSubmit: (sponsorshipId: string) => Promise<void>
+    tokenSymbol: string
+    sponsorships: UndelegateSponsorship[]
+    totalAmount: BN
+}
+
+interface UndelegateSponsorship {
+    id: string
+    streamId: string
+    amount: BN
+    minimumStakingPeriodSeconds: number
+    joinTimestamp: number
+}
+
+function getOptimalSponsorship(
+    sponsorships: UndelegateSponsorship[],
+    requestedAmount: BN,
+): UndelegateSponsorship | undefined {
+    return sponsorships.find(
+        (s) =>
+            hasStakedLongEnough(s.joinTimestamp, s.minimumStakingPeriodSeconds) &&
+            s.amount.isGreaterThanOrEqualTo(requestedAmount),
+    )
+}
+
+function hasStakedLongEnough(joinTimestamp: number, minimumStakingPeriodSeconds: number) {
+    const joinDate = moment(joinTimestamp * 1000)
+    const slashedBeforeDate = joinDate.clone().add(minimumStakingPeriodSeconds, 'seconds')
+    return moment().isAfter(slashedBeforeDate)
+}
+
+export default function ForceUndelegateModal({
+    title = 'Force unstake',
+    onResolve,
+    onSubmit,
+    submitLabel = 'Force unstake',
+    tokenSymbol,
+    sponsorships: unsortedSponsorships,
+    totalAmount,
+    ...props
+}: Props) {
+    const sponsorships = useMemo(
+        () => [...unsortedSponsorships].sort((a, b) => b.amount.comparedTo(a.amount)),
+        [unsortedSponsorships],
+    )
+
+    const [selectedSponsorshipId, setSelectedSponsorshipId] = useState<
+        string | undefined
+    >(
+        getOptimalSponsorship(sponsorships, totalAmount)?.id ??
+            (sponsorships.length > 0 ? sponsorships[0].id : undefined),
+    )
+    const [busy, setBusy] = useState(false)
+
+    const willSlash = useMemo(() => {
+        const sponsorship = sponsorships.find((s) => s.id === selectedSponsorshipId)
+        if (sponsorship) {
+            return !hasStakedLongEnough(
+                sponsorship.joinTimestamp,
+                sponsorship.minimumStakingPeriodSeconds,
+            )
+        }
+        return false
+    }, [selectedSponsorshipId, sponsorships])
+
+    const isPartialPayout = useMemo(() => {
+        const sponsorship = sponsorships.find((s) => s.id === selectedSponsorshipId)
+        if (sponsorship) {
+            return sponsorship.amount.isLessThan(totalAmount)
+        }
+        return false
+    }, [selectedSponsorshipId, sponsorships, totalAmount])
+
+    const canSubmit = selectedSponsorshipId != null
+
+    return (
+        <ForceUndelegateFormModal
+            {...props}
+            title={title}
+            canSubmit={canSubmit && !busy}
+            submitting={busy}
+            submitLabel={submitLabel}
+            onSubmit={async () => {
+                setBusy(true)
+                try {
+                    if (selectedSponsorshipId != null) {
+                        await onSubmit(selectedSponsorshipId)
+                        onResolve?.(selectedSponsorshipId)
+                    }
```

Comment:
If we somehow get here and `selectedSponsorshipId` is nullish then `setBusy` is never reset.

Let's add
```tsx
if (!selectedSponsorshipId) {
    return
}
```
before `setBusy(true)`, or
```tsx
… } finally {
    setBusy(false)
}
```